Source/Program.cs

```diff
@@ -2,31 +2,89 @@
-        ServiceProvider serviceProvider = new ServiceCollection()
-        .AddMyAppServices()
-        .BuildServiceProvider();
-
-        // get services and scope only the dependencies registered for this container
-        using (var scope = serviceProvider.CreateScope())
+        public static void Main()
         {
-            var requiredServices = scope.ServiceProvider.GetServices<ICodeChallenge>()
-            .First(x => x.CodeChallengeSource == CodeChallengeSource.LeetCode && x.GetType().Name == "MinimumOperationsToMakeTheArrayIncreasing");
+            int number = ShowConsoleCommands();
+            if (number != 0)
+            {
+                // configure and register services in container
+                ServiceProvider serviceProvider = new ServiceCollection()
+                 .AddMyAppServices()
+                 .BuildServiceProvider();
+
+                // get services and scope only the dependencies registered for this container
+                using var scope = serviceProvider.CreateScope();
 
-            var result = requiredServices.Execute();
+                var requiredServices = scope.ServiceProvider.GetServices<ICodeChallenge>()
+                .First(x => x.CodeChallengeSource == CodeChallengeSource.LeetCode && x.CodeChallengeNumber == number);
 
-            var concatResult = "";
+                var result = requiredServices.Execute();
 
-            foreach (var innerresult in result)
+                var concatResult = string.Empty;
+
+                // display results in console
+                foreach (var innerresult in result)
+                {
+                    concatResult = string.Join(", ", innerresult);
+                    Console.WriteLine(concatResult);
+                }
+            }
+        }
+
+        private static int ShowConsoleCommands()
+        {
+            bool showCommand = true;
+            int number = default;
+            while (showCommand)
             {
-                concatResult = string.Join(", ", innerresult);
-                Console.WriteLine(concatResult);
+                // show command prompts.
+                string folderPath = $@"{Directory.GetCurrentDirectory()}\..\Common\LeetCode";
+                string[] files = Directory.GetFiles(folderPath, "*.cs");
+
+                Console.WriteLine("List of LeetCode Code Challenges:");
+                Console.WriteLine();
+                foreach (string file in files)
+                {
+                    // get filename without extension
+                    string fileName = Path.GetFileNameWithoutExtension(file);
+
+                    // load the desired assembly
+                    Assembly externalAssembly = Assembly.LoadFrom("../Common/obj/Debug/net6.0/common.dll");
+
+                    Type type = externalAssembly.GetType($"Common.LeetCode.{fileName}");
+                    if (type == null) continue;
+                    object instance = Activator.CreateInstance(type);
```

Comment:
Tip: refactor in the near future.

Reply:
@codebreaker131417 thanks for the tip. Will take care of it next time.